smithy-trait-codegen/src/main/java/software/amazon/smithy/traitcodegen/generators/BuilderGenerator.java

```diff
@@ -0,0 +1,294 @@
+        @Override
+        public Void listShape(ListShape shape) {
+            writeValuesProperty(shape);
+            return null;
+        }
+
+        @Override
+        public Void mapShape(MapShape shape) {
+            writeValuesProperty(shape);
+            return null;
+        }
+
+        @Override
+        public Void structureShape(StructureShape shape) {
+            shape.members().forEach(this::writeProperty);
+            return null;
+        }
+
+        private void writeProperty(MemberShape shape) {
+            Optional<String> builderRefOptional =
+                    symbolProvider.toSymbol(shape).getProperty(SymbolProperties.BUILDER_REF_INITIALIZER, String.class);
+            if (builderRefOptional.isPresent()) {
+                writer.write(BUILDER_REF_TEMPLATE, BuilderRef.class, symbolProvider.toSymbol(shape),
+                        symbolProvider.toMemberName(shape),
+                        builderRefOptional.orElseThrow(RuntimeException::new));
+            } else {
+                writer.write("private $B $L;", symbolProvider.toSymbol(shape),
+                        symbolProvider.toMemberName(shape));
+            }
+        }
+
+        private void writeValuesProperty(Shape shape) {
+            Symbol collectionSymbol = symbolProvider.toSymbol(shape);
+            writer.write(BUILDER_REF_TEMPLATE, BuilderRef.class, collectionSymbol, VALUES,
+                    collectionSymbol.expectProperty(SymbolProperties.BUILDER_REF_INITIALIZER));
+        }
+    }
+
+
+    private final class BuilderSetterGenerator extends ShapeVisitor.Default<Void> {
+        @Override
+        protected Void getDefault(Shape shape) {
+            throw new UnsupportedOperationException("Does not support shape of type: " + shape.getType());
+        }
+
+        @Override
+        public Void listShape(ListShape shape) {
+            shape.accept(new SetterVisitor(VALUES));
+            return null;
+        }
+
+        @Override
+        public Void mapShape(MapShape shape) {
+            shape.accept(new SetterVisitor(VALUES));
+            return null;
+        }
+
+        @Override
+        public Void structureShape(StructureShape shape) {
+            shape.members().forEach(
+                    memberShape -> memberShape.accept(new SetterVisitor(symbolProvider.toMemberName(memberShape))));
+            writer.newLine();
+            return null;
+        }
+    }
+
+
+    private final class SetterVisitor extends ShapeVisitor.Default<Void> {
+        private final String memberName;
+
+        private SetterVisitor(String memberName) {
+            this.memberName = memberName;
+        }
+
+        @Override
+        protected Void getDefault(Shape shape) {
+            writer.openBlock(ACCESSOR_TEMPLATE, "}",
+                    memberName, symbolProvider.toSymbol(shape), () -> {
+                        writer.write("this.$1L = $1L;", memberName);
+                        writer.writeWithNoFormatting(RETURN_THIS);
```

Comment:
The use of static variables was because that string is re-used 9 times within the builderGenerator. In general when I used a string > 3 times in the same class I like to pull out to a static variable.

Reply:
Is size increase what's worries you? Usually the compiler will deduplicate and create a single string in the constant pool. Using a static variable won't help for that.
The argument for using static variables is that you only need to change one place which sometimes overweights the noise it adds, but for simple strings like this one it does not help in that regard.

Reply:
Size increase is not the issue. Just trying to reduce number of places I define that string.